crates/fuel-core/src/graphql_api/worker_service.rs

```diff
@@ -508,7 +508,7 @@ where
             off_chain_height.map(|height| BlockHeight::new(height.saturating_add(1)));
 
         let import_result =
-            import_result_provider.block_event_at_height(next_block_height)?;
+            import_result_provider.block_event_at_height(next_block_height.into())?;
```

Comment:
Is it possible to make `on_chain_database.latest_height()` return `BlockHeightQuery`? 

This makes me think that the name should be something different.

Reply:
We could change `latest_height()` to return `BlockHeightQuery`.  I didn't do this because, IIUC, a database should be able to return None if it can't find the requested key.  Is the genesis block stored in the DB?  I agree the name `BlockHeightQuery` isn't quite right.  The only other name I can think of is `SpecificBlockOrGenesis`